third_party/move/move-compiler-v2/src/bytecode_generator.rs

```diff
@@ -720,7 +719,25 @@ impl<'env> Generator<'env> {
             .env()
             .get_node_instantiation_opt(id)
             .unwrap_or_default();
-        let args = self.gen_arg_list(args);
+        // Function calls can have implicit conversion of &mut to &, need to compute implicit
+        // conversions.
+        let param_types: Vec<Type> = self
+            .env()
+            .get_function(fun)
+            .get_parameters()
+            .into_iter()
+            .map(|Parameter(_, ty)| ty.instantiate(&type_args))
+            .collect();
+        if args.len() != param_types.len() {
+            self.internal_error(id, "inconsistent type arity");
```

Comment:
what is the advantage of this C style errors versus Rust errors?

Reply:
This actually neither or, its an error with source location information added to the compilers global environment, which can contain many of them. Its marked as 'internal' because it is unexpected. Nevertheless will be reported with possible errors at the source.

Reply:
+possible other errors at the source